mypy/checkexpr.py

```diff
@@ -3593,6 +3593,11 @@ def accept(self,
         """
         if node in self.type_overrides:
             return self.type_overrides[node]
+        # Don't use an Any type context, since that will cause more
+        # anys than we want for things that consume the inferred
+        # types.
+        if isinstance(get_proper_type(type_context), AnyType):
+            type_context = None
```

Comment:
This may actually be a big change. I am curious how this behaves w.r.t. to things like
```python
def foo(x: Any) -> None: ...
T = TypeVar('T', bound=int)
def g(x: T) -> T: # also -> Optional[T] and -> List[T], IIRC all three are special cased
    ...
foo(g('no'))
```
If these example don't change behaviour, then I think this is good to go.

Reply:
Doesn't seem to change anything there